tests/test-yamldata.c

```diff
@@ -267,14 +278,35 @@ test_yaml_read_languages (void)
 					"  - locale: en_GB\n"
 					"    percentage: 100\n";
 
-	cpt = as_yaml_test_read_data (yamldata_languages);
+	cpt = as_yaml_test_read_data_ok (yamldata_languages);
 	g_assert_cmpstr (as_component_get_id (cpt), ==, "org.example.Test");
 
 	g_assert_cmpint (as_component_get_language (cpt, "de_DE"), ==, 48);
 	g_assert_cmpint (as_component_get_language (cpt, "en_GB"), ==, 100);
 	g_assert_cmpint (as_component_get_language (cpt, "invalid_C"), ==, -1);
 }
 
+void
+test_yaml_corrupt_data (void)
+{
+	GError *error = NULL;
+	g_autoptr(AsComponent) cpt = NULL;
+	const gchar *yamldata_corrupt = "---\n"
+					"ID: org.example.Test\n"
+					"\007\n";
+
+	cpt = as_yaml_test_read_data (yamldata_corrupt, &error);
+
+	g_assert_error (error, AS_METADATA_ERROR, AS_METADATA_ERROR_FAILED);
+	g_assert_null (cpt);
+
+	if (error != NULL) {
+		g_assert_true (g_str_has_prefix (error->message, "Invalid DEP-11 file found: YAML parsing error"));
```

Comment:
This could be considered too brittle, as it's closely tied to the exact format of the error messages generated.